solr/modules/analysis-extras/src/java/org/apache/solr/update/processor/DocumentCategorizerUpdateProcessorFactory.java

```diff
@@ -0,0 +1,569 @@
+                + "' children '"
+                + PATTERN_PARAM
+                + "' and '"
+                + REPLACEMENT_PARAM
+                + "' are both mandatory and can not be null");
+      }
+      if (!(patt instanceof String && replacement instanceof String)) {
+        throw new SolrException(
+            SERVER_ERROR,
+            "Init param '"
+                + DEST_PARAM
+                + "' children '"
+                + PATTERN_PARAM
+                + "' and '"
+                + REPLACEMENT_PARAM
+                + "' must both be strings (i.e. <str>)");
+      }
+      if (0 != destList.size()) {
+        throw new SolrException(
+            SERVER_ERROR,
+            "Init param '"
+                + DEST_PARAM
+                + "' has unexpected children: '"
+                + destList.getName(0)
+                + "'");
+      }
+
+      try {
+        this.pattern = Pattern.compile(patt.toString());
+      } catch (PatternSyntaxException pe) {
+        throw new SolrException(
+            SERVER_ERROR,
+            "Init param '"
+                + DEST_PARAM
+                + "' child '"
+                + PATTERN_PARAM
+                + " is not a valid regex pattern: "
+                + patt,
+            pe);
+      }
+      dest = replacement.toString();
+
+    } else if (d instanceof String) {
+      dest = d.toString();
+    } else {
+      throw new SolrException(
+          SERVER_ERROR,
+          "Init param '"
+              + DEST_PARAM
+              + "' must either be a string "
+              + "(i.e. <str>) or a list (i.e. <lst>) containing '"
+              + PATTERN_PARAM
+              + "' and '"
+              + REPLACEMENT_PARAM);
+    }
+  }
+
+  @Override
+  public void inform(final SolrCore core) {
+    this.solrHome = Paths.get(core.getCoreContainer().getSolrHome());
+    srcSelector =
+        FieldMutatingUpdateProcessor.createFieldNameSelector(
+            core.getResourceLoader(),
+            core,
+            srcInclusions,
+            FieldMutatingUpdateProcessor.SELECT_NO_FIELDS);
+
+    for (SelectorParams exc : srcExclusions) {
+      srcSelector =
+          FieldMutatingUpdateProcessor.wrap(
+              srcSelector,
+              FieldMutatingUpdateProcessor.createFieldNameSelector(
+                  core.getResourceLoader(),
+                  core,
+                  exc,
+                  FieldMutatingUpdateProcessor.SELECT_NO_FIELDS));
+    }
+  }
+
+  @Override
+  public final UpdateRequestProcessor getInstance(
+      SolrQueryRequest req, SolrQueryResponse rsp, UpdateRequestProcessor next) {
+    final FieldNameSelector srcSelector = getSourceSelector();
+    return new UpdateRequestProcessor(next) {
+
+      DocumentCategorizerDL documentCategorizerDL = null;
+
+      {
+        // Initialize the categorizer.
+        FileStore fs = req.getCoreContainer().getFileStore();
+
+        var path = solrHome.resolve(ClusterFileStore.FILESTORE_DIRECTORY);
```

Comment:
@gerlowskija this is the line I was referenceing, where the below `modelFile` and `vocabFile` feel like they are not using the ClusterFileStore properly....